conda_build/build.py

```diff
@@ -1357,13 +1357,21 @@ def build_info_files_json_v1(m, prefix, files, files_with_prefix):
         }
         if file_info["path_type"] == PathType.hardlink:
             file_info["size_in_bytes"] = os.stat(path).st_size
-        elif not isfile(path):
-            # this is a softlink that points to nowhere, so is zero bytes
-            file_info["size_in_bytes"] = 0
-            warnings.warn('file %s is a symlink with no target' % path, UserWarning)
-        else:
+        elif file_info["path_type"] == PathType.softlink:
+            dest = os.path.normpath(os.path.join(os.path.dirname(path), os.readlink(path)))
             # softlink that points somewhere
-            file_info["size_in_bytes"] = os.stat(path).st_size
+            if isdir(dest) and not islink(dest):
+                file_info["size_in_bytes"] = 0
+            elif isfile(dest) and not islink(dest):
+                file_info["size_in_bytes"] = os.stat(path).st_size
+            elif islink(dest):
+                sys.exit("TODO :: Nested symlinks not handled yet")
```

Comment:
Was this intentional? Why should symlinks to non-existing files not be allowed? This is a breaking change.